samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs

```diff
@@ -11,31 +13,38 @@ namespace TagHelperSample.Web.Components
     [ViewComponent(Name = "FeaturedMovies")]
     public class FeaturedMoviesComponent : ViewComponent
     {
-        private MoviesService _moviesService;
+        private readonly IMemoryCache _cache;
+        private readonly MoviesService _moviesService;
 
-        public FeaturedMoviesComponent(MoviesService moviesService)
+        public FeaturedMoviesComponent(MoviesService moviesService, IMemoryCache cache)
         {
             _moviesService = moviesService;
+            _cache = cache;
         }
 
         public IViewComponentResult Invoke()
         {
-            IExpirationTrigger trigger;
-            var movies = _moviesService.GetFeaturedMovies(out trigger);
-            
-            // Add custom triggers
-            EntryLinkHelpers.ContextLink.AddExpirationTriggers(new[] { trigger });
```

Comment:
Not sure I fully follow. With the previous design, I could add arbitrary expiration triggers to the tag helper without having to create new cache entries.

```
EntryLinkHelpers.ContentLink.AddExpirationTriggers(new[] { FileProvider.Watch("/user.dat") });
```

With this design you're forced to cache something for the linking to work. In this example, @kichalla is creating a `featured_movies` cache entry even though he doesn't actually particularly care about caching it.